gateway/src/useRedirectMap.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+import io from "socket.io-client";
+import { flags } from "../config/flags.js";
+
+function useRedirectMap() {
```

Comment:
This is sick.

Can you explain more the decision of full local gateway vs near-bos-webcomponent? (it was related to websockets, right?)

Reply:
`near-bos-webcomponent` is really great for quickly launching a consumer gateway, but my vision is to have a gateway for developers, and to implement it we need to have direct access to the gateway code. 

Some dev features I think will be useful on a local gateway but not needed in normal gateways:
- Hot reload
- SocialDB/API mocking
- Debuging Tools like [this](https://bos-web-engine.vercel.app/bwe-demos.near/StateAndTrust.Root?isDebug=true&showMonitor=true)